cou/utils/openstack.py

```diff
@@ -147,3 +229,98 @@ def lookup(cls, component: str, version: str) -> List[str]:
             if version in version_range:
                 compatible_os_releases.append(openstack_release)
         return compatible_os_releases
+
+
+def determine_next_openstack_release(release: str) -> Tuple[str, str]:
+    """Determine the next release after the one passed as a str.
+
+    The returned value is a tuple of the form: ('2020.1', 'ussuri')
+
+    :param release: the release to use as the base
+    :type release: str
+    :returns: the release tuple immediately after the current one.
+    :rtype: Tuple[str, str]
+    :raises: ValueError if the current release doesn't actually exist
+    """
+    old_index = list(OPENSTACK_CODENAMES.values()).index(release)
+    new_index = old_index + 1
+    return list(OPENSTACK_CODENAMES.items())[new_index]
```

Comment:
This will raise an IndexError if I do `determine_next_openstack_release("antelope")` and I think we should raise other exception or the `("2023.1", "antelope")`.
Also we are using mostly values from `OPENSTACK_CODENAMES`, so why not switch keys and values here? Or maybe not define it as dict, but as two lists?

Reply:
or even an Enum with a `next()` method (so you can catch being out of range if needed). Very rough sketch (would need to be polished):
```
class OpenStackVersion(Enum):
    ussuri = '2020.1'
    victoria = '2020.2'
    wallaby = '2021.1'

    def next(self):
        cur_index = list(self.__class__).index(self)
        try:
            return list(self.__class__)[cur_index+1]
        except IndexError:
            raise SomeCustomException
```